### PR TITLE
Port to playbin

### PR DESCRIPTION
**Pull request by devmrfitz**
Ported to playbin as mentioned in Issue #15

### CODE REVIEW

File: page.py [hunk 1]
```diff
@@ -205,20 +205,20 @@ def new_target(self):
...
+            GLib.timeout_add(1000, aplay.play, os.path.join(
                     self._sounds_path,
-                    self._media_data[self.target][0]))
+                    self._media_data[self.target][0])[:-1])
         else:
-            play_audio_from_file(os.path.join(
+            aplay.play(os.path.join(
                     self._sounds_path,
-                    self._media_data[self.target][0]))
+                    self._media_data[self.target][0])[:-1])
```
Comment (srevinsaju) [inline page.py]: Can you explain the need for `[:-1]` slicing?

Currently, I am at this error:
```
1608878628.474940 ERROR root: gst-resource-error-quark: Resource not found. (3) ../gstreamer/plugins/elements/gstfilesrc.c(536): gst_file_src_start (): /GstPlayBin:playbin/GstURIDecodeBin:uridecodebin6/GstFileSrc:source:
No such file '/xxx/git/sugar/activities/iknowmyabcs/sounds/es/dado.og'
```

**Review by srevinsaju — CHANGES_REQUESTED** [review]
Christmas Greetings! Thanks for the patch. Minor changes

File: page.py (hunk 1 above)
Comment (devmrfitz) [reply page.py]: I've made some changes. Please recheck

File: page.py [hunk 2]
```diff
@@ -207,18 +207,18 @@ def _play_target_sound(self):
...
+                    self._media_data[self.target][1]).replace('\n', ' ').replace('\r', ''))
             GLib.timeout_add(1000, aplay.play, os.path.join(
                     self._sounds_path,
-                    self._media_data[self.target][0])[:-1])
+                    self._media_data[self.target][0]).replace('\n', ' ').replace('\r', ''))
         else:
             aplay.play(os.path.join(
                     self._sounds_path,
-                    self._media_data[self.target][0])[:-1])
+                    self._media_data[self.target][0]).replace('\n', ' ').replace('\r', ''))
```
Comment (srevinsaju) [inline page.py]: I suppose it would be better if we use `str.strip` to remove non-printable characters like `\n\r`.

**Review by srevinsaju — CHANGES_REQUESTED** [review]

File: page.py (hunk 1 above)
Comment (srevinsaju) [reply page.py]: Re-reviewed

File: page.py (hunk 2 above)
Comment (devmrfitz) [reply page.py]: Please have a look now

Comment (srevinsaju) [reply page.py]: Cool, its better now!

**Comment by srevinsaju** [comment]
Thanks for the changes, I have created https://github.com/devmrfitz/iknowmyabcs/pull/1, kindly test, review and merge.

**Review by srevinsaju — APPROVED** [review]
@devmrfitz Perfect! Good to merge!

**Comment by quozl** [comment]
Thanks.  Reviewed.  Some changes made.  Tested.  Merged by hand.

Here are the changes;

```diff
diff --git a/README.md b/README.md
index fc6b757..dabcf27 100644
--- a/README.md
+++ b/README.md
@@ -19,3 +19,10 @@ How to upgrade?
 On Sugar desktop systems;
 * use [My Settings](https://help.sugarlabs.org/my_settings.html), [Software Update](https://help.sugarlabs.org/my_settings.html#software-update), or;
 * use Browse to open [activities.sugarlabs.org](https://activities.sugarlabs.org/), search for `ABCs`, then download.
+
+How to integrate?
+=================
+
+I Know My ABCs depends on Python, the Sugar Toolkit, PyGObject bindings for GTK 3, and GStreamer.
+
+I Know My ABCs is started by [Sugar](https://github.com/sugarlabs/sugar).
diff --git a/aplay.py b/aplay.py
index fc98999..5995a2e 100644
--- a/aplay.py
+++ b/aplay.py
@@ -3,7 +3,7 @@
 #
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
-# the Free Software Foundation; either version 2 of the License, or
+# the Free Software Foundation; either version 3 of the License, or
 # (at your option) any later version.
 #
 # This program is distributed in the hope that it will be useful,
@@ -68,4 +68,4 @@ class Aplay:
         self._pipeline = None
 
 
-aplay = Aplay()
\ No newline at end of file
+aplay = Aplay()
diff --git a/page.py b/page.py
index a61ad14..727ab57 100644
--- a/page.py
+++ b/page.py
@@ -12,13 +12,13 @@
 # Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 # Boston, MA 02111-1307, USA.
 
-import gi
-gi.require_version('Gst','1.0')
 from gi.repository import Gtk, Gdk, GLib, GdkPixbuf
 import os
 from random import uniform
+
 from gettext import gettext as _
 
+from aplay import aplay
 
 import logging
 _logger = logging.getLogger('iknowmyabcs-activity')
@@ -29,7 +29,7 @@ GRID_CELL_SIZE = style.GRID_CELL_SIZE
 
 from genpieces import generate_card, genblank
 from utils.sprites import Sprites, Sprite
-from aplay import aplay
+
 
 XDIM = 6
 YDIM = 5
@@ -207,18 +207,18 @@ class Page():
         if self._activity.mode in ['letter', 'find by letter']:
             aplay.play(os.path.join(
                     self._sounds_path,
-                    self._media_data[self.target][1]).strip())
+                    self._media_data[self.target][1]))
         elif self._activity.mode == 'picture':
             aplay.play(os.path.join(
                     self._sounds_path,
-                    self._media_data[self.target][1]).strip())
-            GLib.timeout_add(1000, aplay.play, os.path.join(
+                    self._media_data[self.target][1]))
+            aplay.play(os.path.join(
                     self._sounds_path,
-                    self._media_data[self.target][0]).strip())
+                    self._media_data[self.target][0]))
         else:
             aplay.play(os.path.join(
                     self._sounds_path,
-                    self._media_data[self.target][0]).strip())
+                    self._media_data[self.target][0]))
         self.timeout = None
 
     def _button_press_cb(self, win, event):
@@ -316,7 +316,7 @@ class Page():
         f = open(path)
         for line in f:
             if len(line) > 0 and line[0] not in '#\n':
-                words = line.split(', ')
+                words = line.strip().split(', ')
                 self._card_data.append([words[0],
                                         words[1].replace('-', ', ')])
                 if words[2].count('#') > 1:
diff --git a/utils/play_audio.py b/utils/play_audio.py
deleted file mode 100644
index f505bec..0000000
--- a/utils/play_audio.py
+++ /dev/null
@@ -1,78 +0,0 @@
(delete the file)
```

Let me know if you have questions.

**Pull request closed without merging**